react/src/components/Table/Table.tsx

```diff
@@ -520,8 +514,8 @@ const Table: React.FC<TableProps> = ({ variantData }) => {
                         variant="primary"
                         onClick={() => {
                             downloadCsv(
-                                formatDataForCsv(rows),
-                                visibleColumns.map(c => c.id)
+                                rows.map(r => r.values as ResultTableColumns),
```

Comment:
I think we'd need to filter out the dummy columns (`emtpyVariationDetails`, `emptyCaseDetails`) as well so user doesn't have those columns in the CSV when they are expanded.